File: Assets/Scripts/Main/PrevDoor.js
#pragma strict

var doorId : int;
var door : GameObject [];

function Start () {
	if(PlayerPrefs.GetInt("ExitDoor") != -1)
		doorId = PlayerPrefs.GetInt("ExitDoor");
	else
		doorId = PlayerPrefs.GetInt("Door");
	
	if(doorId != 0) {
		var obj : GameObject = Instantiate(door[doorId-1]);
		obj.transform.parent = transform;
		obj.transform.localPosition = Vector3.zero;
		obj.transform.localRotation = Quaternion.identity;
	}
}

File: Assets/Scripts/Main/PrevDoor.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class PrevDoor : MonoBehaviour
{
    public int doorId;
    public GameObject[] door;
    public virtual void Start()
    {
        if (PlayerPrefs.GetInt("ExitDoor") != -1)
        {
            this.doorId = PlayerPrefs.GetInt("ExitDoor");
        }
        else
        {
            this.doorId = PlayerPrefs.GetInt("Door");
        }
        if (this.doorId != 0)
        {
            GameObject obj = UnityEngine.Object.Instantiate(this.door[this.doorId - 1]);
            obj.transform.parent = this.transform;
            obj.transform.localPosition = Vector3.zero;
            obj.transform.localRotation = Quaternion.identity;
        }
    }

}